### ISSUE TITLE
[Combobox] For Multiple Group Selected and Groups with Group Selected keyboard user can select one option more then once

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3447]
### 😯 Current Behavior

https://github.com/user-attachments/assets/24cecbfd-de1a-44d7-bd02-d07bf62c349a

### 🤔 Expected Behavior/possible solution?

When user selects the option their focus should be automatically placed on that selected option and pressing `Enter` once again should deselect that option.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/form-combobox-multiple--group-selected
2. Select any option using only keyboard
3. Without moving after selecting option press `Enter` again
4. Check if selected option is deselected or if another tag with option's name was added to combobox

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-combobox-multiple--group-selected

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.1.9

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
